x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.test.ts

```diff
@@ -0,0 +1,658 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { BulkIndexDocumentsParams, SearchParams } from 'elasticsearch';
+import { savedObjectsClientMock } from 'src/core/server/mocks';
+
+import { AlertServices } from '../../../../../alerting/server/types';
+import { Logger } from '../../../../../../../../src/core/server';
+import {
+  buildBulkBody,
+  singleBulkIndex,
+  singleSearchAfter,
+  searchAfterAndBulkIndex,
+} from './utils';
+import { SignalHit } from '../../types';
+import {
+  sampleDocNoSortId,
+  sampleSignalAlertParams,
+  sampleDocSearchResultsNoSortId,
+  sampleDocSearchResultsNoSortIdNoHits,
+  sampleDocSearchResultsWithSortId,
+  sampleEmptyDocSearchResults,
+  repeatedSearchResultsWithSortId,
+} from './__mocks__/es_results';
+import { SignalAlertParams } from './types';
+import { buildEventsSearchQuery } from './build_events_query';
+
+const mockLogger: Logger = {
+  info: jest.fn((logString: string) => {
+    return logString;
+  }),
+  warn: jest.fn((logString: string) => {
+    return logString;
+  }),
+  trace: jest.fn(),
+  debug: jest.fn((logString: string) => {
+    return logString;
+  }),
+  error: jest.fn((logString: string) => {
+    return logString;
+  }),
+  fatal: jest.fn(),
+  log: jest.fn(),
+};
```

Comment:
Searching through the code base I think this compact form is most of what the other code writers in other sections of Kibana use for mock logger:

```ts
const mockLogger: Logger = {
  log: jest.fn(),
  trace: jest.fn(),
  debug: jest.fn(),
  info: jest.fn(),
  warn: jest.fn(),
  error: jest.fn(),
  fatal: jest.fn(),
};
```

Testing it and it seems to work well here.